app/views/split_checkout/payment/_stripe_sca.html.haml

```diff
@@ -1,15 +1,17 @@
+- saved_credit_cards = spree_current_user&.credit_cards&.with_payment_profile.to_a
```

Comment:
I would argue it would be better to keep `load_saved_credit_cards` (with out setting the `@selected_card` probably) in the controller/concern. To me it feels wrong as the view is essentially doing a query, which should be the responsibility of the controller. That said, it's more an issue with rails view system allowing you to do that sort of things.

Reply:
So these little changes around loading/assigning credit cards and loading/assigning the voucher object are related. The issue is that both the checkout controller and the vouchers controller need to potentially render some of these views as it relates to vouchers. The view doesn't work unless the cards are loaded, but if we do it at the controller level then essentially we need to write the exact same code in multiple controllers. So it turns out this logic isn't really specific to either controller, but specific to the view. Know what I mean? So what I wanted to do was avoid duplicating that logic in both controllers, essentially both of them would have to do something like this:
```ruby
@saved_credit_cards = <load the cards>
@voucher_adjustment = <load the voucher>
```
So moving these bits around and actually moving them _out_ of the controller level means the view has no external dependencies and it becomes a lot more portable in terms of being able to render it more easily from different contexts.

It does look a bit funky sitting in the view. I guess the more idiomatic way of doing this would be to extract it to a method in a view helper. Maybe that's preferable..? I think I avoided doing that for now just because currently it's only used one place, so the method would only have one caller. What do you think?

Reply:
I can see what you are trying to do but it still doesn't sit well with me. I don't think the view helper is the solution either, you would just be moving the problem around, the view would still be making the query, it's just moved away from the view file.

The only thing I can think of would be to add the data loading to their own concern which would need to be included in the related controllers. But it does keep the view external dependency.

I guess loading the data in the view is an acceptable trade off, but I can't help wondering if it's a bit of a premature optimisation, are these views going be to be render in many different context ?